source/v5/mqtt5_to_mqtt3_adapter.c

```diff
@@ -326,6 +340,13 @@ static struct aws_mqtt_adapter_connect_task *s_aws_mqtt_adapter_connect_task_new
     connect_task->clean_session = connection_options->clean_session;
 
     return connect_task;
+
+error:
+    aws_ref_count_release(&adapter->internal_refs);
```

Comment:
Instead of doing this, let's move the acquire after the last control flow that can error out.  Having to release refs in error blocks is something to avoid if possible.

Reply:
The task was using the `adapter->allocator`. If we do not acquire the adapter beforehand, would there be chance that that the allocator get released with the adapter?

Reply:
no